diff --git a/go-reuseport/poll/poll_linux.go b/go-reuseport/poll/poll_linux.go
index 2f22617ed..de452f71a 100644
--- a/go-reuseport/poll/poll_linux.go
+++ b/go-reuseport/poll/poll_linux.go
@@ -7,7 +7,7 @@ import (
 	"sync"
 	"unknown/sys/unix"
 
-	"gx/ipfs/QmSGRM5Udmy1jsFBr1Cawez7Lt7LZ3ZKA23GGVEsiEW6F3/eventfd"
+	"unknown/eventfd"
 )
 
 type Poller struct {
diff --git a/go-reuseport/singlepoll/linux.go b/go-reuseport/singlepoll/linux.go
index a59ec7de0..80c1b79fe 100644
--- a/go-reuseport/singlepoll/linux.go
+++ b/go-reuseport/singlepoll/linux.go
@@ -8,8 +8,8 @@ import (
 	"sync"
 	"unknown/sys/unix"
 
-	"gx/ipfs/QmSGRM5Udmy1jsFBr1Cawez7Lt7LZ3ZKA23GGVEsiEW6F3/eventfd"
-	logging "gx/ipfs/QmSpJByNKFX1sCsHBEp3R73FL4NF6FnQTEGyNAXHm2GS52/go-log"
+	"unknown/eventfd"
+	logging "unknown/go-log"
 )
 
 var (
