diff --git a/go-ipfs-cmdkit/files/is_hidden_windows.go b/go-ipfs-cmdkit/files/is_hidden_windows.go
index fdb85a369..7679433df 100644
--- a/go-ipfs-cmdkit/files/is_hidden_windows.go
+++ b/go-ipfs-cmdkit/files/is_hidden_windows.go
@@ -6,7 +6,7 @@ import (
 	"path/filepath"
 	"strings"
 
-	windows "gx/ipfs/QmPXvegq26x982cQjSfbTvSzZXn7GiaMwhhVPHkeTEhrPT/sys/windows"
+	windows "golang.org/x/sys/windows"
 )
 
 func IsHidden(f File) bool {
diff --git a/vendor/unknown/go-sockaddr/net/net_windows.go b/vendor/unknown/go-sockaddr/net/net_windows.go
index 73d686ed7..9a65cf931 100644
--- a/vendor/unknown/go-sockaddr/net/net_windows.go
+++ b/vendor/unknown/go-sockaddr/net/net_windows.go
@@ -1,7 +1,7 @@
 package sockaddrnet
 
 import (
-	"gx/ipfs/QmPXvegq26x982cQjSfbTvSzZXn7GiaMwhhVPHkeTEhrPT/sys/windows"
+	"golang.org/x/sys/windows"
 )
 
 const (
diff --git a/vendor/unknown/go4-lock/lock_windows.go b/vendor/unknown/go4-lock/lock_windows.go
index 1a83cef28..425748790 100644
--- a/vendor/unknown/go4-lock/lock_windows.go
+++ b/vendor/unknown/go4-lock/lock_windows.go
@@ -22,7 +22,7 @@ import (
 	"os"
 	"sync"
 
-	"gx/ipfs/QmPXvegq26x982cQjSfbTvSzZXn7GiaMwhhVPHkeTEhrPT/sys/windows"
+	"golang.org/x/sys/windows"
 )
 
 func init() {
